p2p/raintree/testutil.go

```diff
@@ -2,12 +2,30 @@
 
 package raintree
 
-import libp2pNetwork "github.com/libp2p/go-libp2p/core/network"
+import (
+	libp2pNetwork "github.com/libp2p/go-libp2p/core/network"
+	"github.com/regen-network/gocuke"
+
+	"github.com/pokt-network/pocket/internal/testutil"
+	typesP2P "github.com/pokt-network/pocket/p2p/types"
+)
 
 // RainTreeRouter exports `rainTreeRouter` for testing purposes.
 type RainTreeRouter = rainTreeRouter
 
+type routerHandlerProxyFactory = testutil.ProxyFactory[typesP2P.MessageHandler]
+
 // HandleStream exports `rainTreeRouter#handleStream` for testing purposes.
 func (rtr *rainTreeRouter) HandleStream(stream libp2pNetwork.Stream) {
-	rtr.handleStream(stream)
+	rtr.UnicastRouter.HandleStream(stream)
+}
+func (rtr *rainTreeRouter) HandlerProxy(
+	t gocuke.TestingT,
+	handlerProxyFactory routerHandlerProxyFactory,
+) {
+	t.Helper()
+
+	// pass original handler to proxy factory & replace it with the proxy
```

Comment:
Can you expand on this? I don't fully understand the need for this (without spending time diving deeper).

Reply:
Apologies for this garbage. It should've been cleaned up, we don't need this. :broom: